### PR TITLE
Fixed #33639 -- Enabled cached template loader in development.

### PR DESCRIPTION
**Pull request by carltongibson**
ticket-33639

### CODE REVIEW

**Comment by carltongibson** [comment]
Hi @collinanderson, @ngnpope, @kezabelle — Follow-up to our discussion on https://github.com/django/django/pull/15140. Can I ask you to give it play and see if there are any blockers? Thanks! 🎁

**Comment by collinanderson** [comment]
@carltongibson I'll use a the branch for a while in development. Seems to be working well so far. Thanks.

File: docs/ref/templates/api.txt [hunk 1]
```diff
@@ -99,7 +99,7 @@ overridden by what's passed by
       * ``'django.template.loaders.app_directories.Loader'`` if and only if
         ``app_dirs`` is ``True``.
 
-      If ``debug`` is ``False``, these loaders are wrapped in
+      These loaders are then wrapped in
```
Comment (felixxm) [inline docs/ref/templates/api.txt:102]: This is far away from the `versionchanged` annotation, so I would add another one here.

File: docs/ref/templates/api.txt [hunk 2]
```diff
@@ -905,10 +905,9 @@ loaders that come with Django:
 
 .. class:: cached.Loader
 
-    By default (when :setting:`DEBUG` is ``True``), the template system reads
-    and compiles your templates every time they're rendered. While the Django
-    template system is quite fast, the overhead from reading and compiling
-    templates can add up.
+    While the Django template system is quite fast, if it needs to read and
+    compile your templates every time they're rendered, the overhead from
+    that can add up.
```
Comment (felixxm) [inline docs/ref/templates/api.txt:910]:
```suggestion
    compile your templates every time they're rendered, the overhead from that
    can add up.
```

File: docs/releases/4.1.txt [hunk 3]
```diff
@@ -342,6 +342,11 @@ Templates
 * :tfilter:`json_script` template filter now allows wrapping in a ``<script>``
   tag without the HTML ``id`` attribute.
 
+* The :class:`cached template loader <django.template.loaders.cached.Loader>`
+   is now enabled in development, when :setting:`DEBUG` is ``True``, and
+   :setting:`OPTIONS['loaders'] <TEMPLATES-OPTIONS>` isn't specified. You may
+   specify ``OPTIONS['loaders']`` to override this, if necessary.
```
Comment (felixxm) [inline docs/releases/4.1.txt:348]: Extra space is unnecessary:
```suggestion
  is now enabled in development, when :setting:`DEBUG` is ``True``, and
  :setting:`OPTIONS['loaders'] <TEMPLATES-OPTIONS>` isn't specified. You may
  specify ``OPTIONS['loaders']`` to override this, if necessary.
```
image

Maybe this should be in the _"Backwards incompatible changes in 4.1"_ section :thinking:. I'd at least add another note there.

File: tests/template_backends/test_django.py [hunk 4]
```diff
@@ -176,27 +176,28 @@ def test_autoescape_default(self):
...
-    @override_settings(DEBUG=True)
-    def test_debug_default_template_loaders(self):
-        engine = DjangoTemplates(
-            {"DIRS": [], "APP_DIRS": True, "NAME": "django", "OPTIONS": {}}
-        )
-        self.assertEqual(engine.engine.loaders, self.default_loaders)
+    def test_default_template_loaders(self):
+        """
+        The cached template loader is alwasys enabled by default.
+        """
```
Comment (felixxm) [inline tests/template_backends/test_django.py:182]:
```suggestion
        """The cached template loader is always enabled by default."""
```

File: tests/template_backends/test_django.py [hunk 5]
```diff
@@ -176,27 +176,28 @@ def test_autoescape_default(self):
...
+        for debug in (True, False):
+            with self.subTest(DEBUG=debug):
+                with override_settings(DEBUG=debug):
```
Comment (felixxm) [inline tests/template_backends/test_django.py:185]: We can keep them together
```suggestion
            with self.subTest(DEBUG=debug), self.settings(DEBUG=debug):
```

**Review by felixxm — COMMENTED** [review]
@carltongibson Thanks :+1: I left small suggestions.

**Comment by carltongibson** [comment]
Thanks for checking @felixxm! Updated. 👍

File: docs/ref/templates/api.txt [hunk 6]
```diff
@@ -99,9 +99,14 @@ overridden by what's passed by
...
         ``app_dirs`` is ``True``.
 
-      If ``debug`` is ``False``, these loaders are wrapped in
+      These loaders are then wrapped in
       :class:`django.template.loaders.cached.Loader`.
 
+      .. versionchanged:: 4.1
+
+        Previously the cached template loader was only enabled by default when
+        ``DEBUG`` was ``False``.
```
Comment (felixxm) [inline docs/ref/templates/api.txt:108]:
```suggestion
        In older versions, the cached template loader was only enabled by
        default when ``DEBUG`` was ``False``.
```

File: docs/releases/4.1.txt [hunk 7]
```diff
@@ -342,6 +342,11 @@ Templates
 * :tfilter:`json_script` template filter now allows wrapping in a ``<script>``
   tag without the HTML ``id`` attribute.
 
+* The :class:`cached template loader <django.template.loaders.cached.Loader>`
+  is now enabled in development, when :setting:`DEBUG` is ``True``, and
+  :setting:`OPTIONS['loaders'] <TEMPLATES-OPTIONS>` isn't specified. You may
+  specify ``OPTIONS['loaders']`` to override this, if necessary.
```
Comment (felixxm) [inline docs/releases/4.1.txt:348]: I think you missed my [note](https://github.com/django/django/pull/15586#discussion_r849326930) about _"Backwards incompatible changes in 4.1"_ section.

**Review by felixxm — APPROVED** [review]
@carltongibson Thanks :+1:

**Pull request merged**